Fabric.Authorization.Persistence.SqlServer/Stores/SqlServerRoleStore.cs

```diff
@@ -118,15 +142,101 @@ public Task<IEnumerable<Role>> GetRoles(string grain, string securableItem = nul
 
             if (!string.IsNullOrWhiteSpace(securableItem))
             {
-                roles = roles.Where(p => string.Equals(p.SecurableItem.Name, securableItem));
+                roles = roles.Where(r => string.Equals(r.SecurableItem.Name, securableItem));
             }
 
             if (!string.IsNullOrWhiteSpace(roleName))
             {
-                roles = roles.Where(p => string.Equals(p.Name, roleName));
+                roles = roles.Where(r => string.Equals(r.Name, roleName));
+            }
+
+            return roles;
+        }
+
+        public async Task<Role> AddPermissionsToRole(Role role, ICollection<Permission> permissions)
+        {
+            var roleEntity = await _authorizationDbContext.Roles
+                .SingleOrDefaultAsync(r =>
+                    r.RoleId == role.Id
```

Comment:
Do we need to include the list of permissions here so we can verify that the permission hasn't already been added to the role? Or is the check being done in the service?

Reply:
The service is doing the check (I left it there so it would be done for all stores).